TRLevelControl/Model/Common/FloorData/FDControl.cs

```diff
@@ -129,7 +129,27 @@ public List<FDTriggerEntry> GetTriggers(FDTrigAction action, int parameter = -1)
             .ToList();
     }
 
+    public List<FDTriggerEntry> GetSwitchKeyTriggers(int entityIndex)
+    {
+        return FindAll(e => e is FDTriggerEntry trigger
+                && (trigger.TrigType == FDTrigType.Switch || trigger.TrigType == FDTrigType.Key)
+                && trigger.SwitchOrKeyRef == entityIndex)
+            .Cast<FDTriggerEntry>()
+            .ToList();
```

Comment:
Might not want to do it now but you can use `OfType` for this type check. Might need to be inside FindAll

Reply:
Thank you. Added this, along with a test, which I forgot to do before.